### PR TITLE
WEBAPP-230: Dynamic page titles

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://integreat.atlassian.net/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

Make our page titles dynamic with react-helmet
Rename ExtraModel name to title
Rename I18nRedirect to I18nRedirectPage

### CODE REVIEW

File: src/locales.json [hunk 1]
```diff
@@ -164,34 +164,35 @@
...
-      "creatingPdf": null,
-      "downloadPdfAt": null,
-      "page": {
-        "loadingFailed": null
-      }
+      "extras": "اضافات",
+      "news": "تازه​ها",
+      "landing": "تو کجایی؟",
+      "search": "جستجو",
+      "disclaimer": "مجوز ‎و اطلاعات تماس"
```
Comment (Schedulaar) [inline src/locales.json]: Hmm... It's not clear from the name of these constants where they are used and what 'functionality' they have... Maybe you could rename them to sth like "disclaimerTitle" or similar...

File: src/modules/app/containers/Helmet.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          return t('extras')
+        }
+        break
+      case SEARCH_ROUTE:
+        return t('search')
+      case MAIN_DISCLAIMER_ROUTE:
+        return t('disclaimer')
+      case LANDING_ROUTE:
+        return t('landing')
+    }
```
Comment (Schedulaar) [inline src/modules/app/containers/Helmet.js]: I don't like that we have another huge "switcher" here... Imo, the routes should decide on their title themselves.
Maybe we can expand the objects in /modules/app/routes/ for that purpose?
If we add an appropriate flow-type there, we should also be safe, that we haven't forgot to set a title to some route.
Also the tests wouldn't bloat a single file, like here Helmet.spec.js, but instead be spread to all of the routes...

File: src/routes/extras/containers/ExtrasPage.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ export class ExtrasPage extends React.Component<Props> {
   getTileModels (): Array<TileModel> {
     return this.props.extras.map(extra => new TileModel({
       id: extra.alias,
-      name: extra.name,
+      name: extra.title,
```
Comment (Schedulaar) [inline src/routes/extras/containers/ExtrasPage.js]: then you could also rename `name` to `title` in the TileModel, right?

Comment (klinzo) [reply src/routes/extras/containers/ExtrasPage.js]: Sure :D

**Pull request merged**